examples/mir_demo_server/server_example_input_device_config.cpp

```diff
@@ -55,133 +74,227 @@ void me::add_input_device_configuration_options_to(mir::Server& server)
     {
-        if (val == touchpad_scroll_mode_edge)
-            return mir_touchpad_scroll_mode_edge_scroll;
-        if (val == touchpad_scroll_mode_two_finger)
-            return mir_touchpad_scroll_mode_two_finger_scroll;
-        return mir_touchpad_scroll_mode_none;
+        if (opt_val)
+        {
+            if (*opt_val == touchpad_scroll_mode_none)
+            {
+                return mir_touchpad_scroll_mode_none;
+            }
+            else if (*opt_val == touchpad_scroll_mode_edge)
+            {
+                return mir_touchpad_scroll_mode_edge_scroll;
+            }
+            else if (*opt_val == touchpad_scroll_mode_two_finger)
+            {
+                return mir_touchpad_scroll_mode_two_finger_scroll;
+            }
+            else if (*opt_val == touchpad_scroll_mode_button_down_scroll)
+            {
+                return mir_touchpad_scroll_mode_button_down_scroll;
+            }
+            else
+            {
+#ifdef __cpp_lib_format
+                throw mir::AbnormalExit{std::format("Unrecognised option for {}: {}", touchpad_scroll_mode_opt, *opt_val)};
+#else
+                throw mir::AbnormalExit{"Unrecognised option for "s + touchpad_scroll_mode_opt + ": "s + *opt_val};
+#endif
+            }
+        }
+        else
+        {
+            return std::nullopt;
+        }
     };
 
-    auto to_profile = [](std::string const& val)
+    auto to_acceleration_profile = [](std::optional<std::string> const& opt_val)-> std::optional<MirPointerAcceleration>
     {
-        if (val == acceleration_none)
-            return mir_pointer_acceleration_none;
-        return mir_pointer_acceleration_adaptive;
+        if (opt_val)
+        {
+            if (*opt_val == acceleration_none)
+            {
+                return mir_pointer_acceleration_none;
+            }
+            else if (*opt_val == acceleration_adaptive)
+            {
+                return mir_pointer_acceleration_adaptive;
+            }
+            else
+            {
+#ifdef __cpp_lib_format
+                throw mir::AbnormalExit{std::format("Unrecognised option for {}: {}", mouse_acceleration_opt, *opt_val)};
+#else
+                throw mir::AbnormalExit{"Unrecognised option for "s + mouse_acceleration_opt + ": "s + *opt_val};
+#endif
+            }
+        }
+        else
+        {
+            return std::nullopt;
+        }
     };
 
-    auto convert_to_click_mode = [](std::string const& val)
+    auto convert_to_click_mode = [](std::optional<std::string> const& opt_val)
+    -> std::optional<MirTouchpadClickMode>
     {
-        if (val == touchpad_click_mode_finger_count)
-            return mir_touchpad_click_mode_finger_count;
-        if (val == touchpad_click_mode_area)
-            return mir_touchpad_click_mode_area_to_click;
-        return mir_touchpad_click_mode_none;
+        if (opt_val)
+        {
+            auto val = *opt_val;
+            if (val == touchpad_click_mode_none)
```

Comment:
Nit: This also might be easier to read as a switch

Reply:
Also true, but...